ext/simplexml/simplexml.c

```diff
@@ -2005,7 +2005,8 @@ PHP_METHOD(SimpleXMLElement, current)
 	}
 
 	if (Z_ISUNDEF(sxe->iter.data)) {
-		return; /* return NULL */
+		zend_throw_error(NULL, "Iterator has already been consumed!");
```

Comment:
Exclamation mark here is a bit unusual :)

I think we might want to make this `Iterator not initialized or already consumed`, as it would also happen before a call to `->rewind()`.

Reply:
> Exclamation mark here is a bit unusual :)

Haha, sorry, bad habits.

> I think we might want to make this Iterator not initialized or already consumed

Makes sense, I'll fix this later.

Reply:
Fixed! Although we still have to decide if this can go into 8.0

Reply:
I think in this case it's best not to touch PHP 8.0.

Reply:
Yeah, makes perfect sense to me